File: truss-compiler-maven-plugin/src/main/java/truss/compiler/generator/syntax/xml/SyntaxEnum.java
package truss.compiler.generator.syntax.xml;

import javax.xml.bind.annotation.*;
import java.util.ArrayList;
import java.util.List;

@XmlRootElement(name = "enum")
@XmlAccessorType(XmlAccessType.FIELD)
public class SyntaxEnum {
    @XmlAttribute
    private String name;
    @XmlElement(name = "property")
    private List<SyntaxProperty> properties = new ArrayList<>();

    public String getName() {
        return name;
    }

    public List<SyntaxProperty> getProperties() {
        return properties;
    }
}

File: Truss.Compiler.Generate/Syntax/Xml/SyntaxEnum.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml.Serialization;

namespace Truss.Compiler.Generate.Syntax.Xml {
    [XmlRoot("enum")]
    public class SyntaxEnum {
        [XmlAttribute("name")]
        public string Name { get; set; }

        [XmlElement("property")]
        public List<SyntaxProperty> Properties { get; set; }
    }
}
